theano/tensor/nnet/abstract_conv.py

```diff
@@ -1484,6 +1505,54 @@ def conv3d_grad_wrt_weights(input,
     return gradWeight_op(input, output_grad, filter_shape[-3:])
 
 
+def dilated_causal_conv(input,
+                        filters,
+                        filter_shape,
+                        input_shape=None,
+                        subsample=1,
+                        filter_flip=True,
+                        filter_dilation=1,
+                        num_groups=1):
+
+    input = as_tensor_variable(input)
+    filters = as_tensor_variable(filters)
+
+    if input.ndim != 3:
+        raise ValueError('Input should be 3D for Dilated Causal convolution.')
+    if filters.ndim != 3:
+        raise ValueError('Filters should be 3D for Dilated Causal convolution')
+
+    input = input.dimshuffle(0, 1, 2, 'x')
+    filters = filters.dimshuffle(0, 1, 2, 'x')
+
+    if input_shape is not None:
+        assert(len(input_shape) == 3)
+        input_shape = tuple(input_shape)
+        input_shape += (1,)
+
+    assert(len(filter_shape) == 3)
+    filter_shape = tuple(filter_shape)
+    filter_shape += (1,)
```

Comment:
I think it would be easy to add support for filter_shape to be simbolic. Just something like:
if isinstance(filter_shape, tuple):
   filter_shape = ...
else:
   filter_shape = theano.tensor.concatenate([[1,], v.shape]).type

Reply:
But that wouldn't work with any of the convolution ops, right? The padding/border mode is passed to `__init__` and if I understood things correctly you can't use Theano variables in that way? I thought they had to be inputs for `make_node` to be able to do that.

My impression was that adding support for symbolic padding would require a fair number of changes. The padding would need to be converted from an `__init__` parameter to an input for each of the convolution ops. But then border mode strings such as `valid`, `full` etc. wouldn't fit inside a Theano variable and those would have to stay in `__init__`, so perhaps you'd need a new `border_mode="custom"` to indicate when the symbolic padding should be used.

Reply:
right. It is more complicated then my simple line.